app/controllers/content_items_controller.rb

```diff
@@ -65,23 +65,43 @@ def redirect_canonical(content_item)
   end
 
   def can_view(item)
-    if auth_bypass_id_header.present?
-      item.includes_auth_bypass_id?(auth_bypass_id_header)
+    if valid_user?(auth_user_id)
+      if item.access_limited?
+        item.viewable_by_bypass_id?(auth_bypass_id) || item.viewable_by_user_id?(auth_user_id) || item.viewable_by_org_id?(auth_org_id)
+      else
+        true
+      end
     else
-      !invalid_user_id? && item.viewable_by?(authenticated_user_uid)
+      false
     end
   end
```

Comment:
This seems to group the behaviour of auth_bypass into this, which I think would actually prevent people from having access.

I think you probably want the conditionals laid out like:

```
if auth bypass header is present
  check whether it's the one for this content
else if item has org or user access limit
  user is in user list or org is in org list
else
  true
end
```

does that make sense?

Reply:
Yup - we've had a go at writing the `viewable?`  method in the content item model.